Mage.Sets/src/mage/cards/d/DescendantsFury.java

```diff
@@ -0,0 +1,196 @@
+        Ability ability = new DealCombatDamageControlledTriggeredAbility(
+                Zone.BATTLEFIELD,
+                new DoIfCostPaid(
+                        new DescendantsFuryEffect(),
+                        new DescendantsFurySacrificeCost()
+                ),
+                true
+        );
+
+        ability.addWatcher(new DamageThisCombatWatcher());
+        this.addAbility(ability);
+    }
+
+    private DescendantsFury(final DescendantsFury card) {
+        super(card);
+    }
+
+    @Override
+    public DescendantsFury copy() {
+        return new DescendantsFury(this);
+    }
+}
+
+class DescendantsFurySacrificeCost extends CostImpl implements SacrificeCost {
+    DescendantsFurySacrificeCost() {
+        this.text = "sacrifice one of them";
+    }
+
+    private DescendantsFurySacrificeCost(final DescendantsFurySacrificeCost cost) {
+        super(cost);
+    }
+
+    @Override
+    public DescendantsFurySacrificeCost copy() {
+        return new DescendantsFurySacrificeCost(this);
+    }
+
+    @Override
+    public boolean pay(Ability ability, Game game, Ability source, UUID controllerId, boolean noMana, Cost costToPay) {
+        DamageThisCombatWatcher watcher = game.getState().getWatcher(DamageThisCombatWatcher.class);
+        if (watcher == null) {
+            return false;
+        }
+        TargetPointer targetPointer = source.getEffects().get(0).getTargetPointer();
+        if (targetPointer == null) {
+            return false;
+        }
+        Player damagedPlayer = game.getPlayer(targetPointer.getFirst(game, source));
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null || damagedPlayer == null) {
+            return false;
+        }
+
+        List<Permanent> possibleSacrifice =
+                watcher.getPermanents(controller.getId(), damagedPlayer.getId())
+                       .stream()
+                       .map(p -> p.getPermanent(game))
+                       .filter(p -> p != null && controller.canPaySacrificeCost(p, source, controllerId, game))
+                       .collect(Collectors.toList());
+
+        if (possibleSacrifice.isEmpty()) {
+            return false;
+        }
+
+        FilterControlledPermanent filter = new FilterControlledPermanent();
+        filter.add(new PermanentInListPredicate(possibleSacrifice));
+
+        TargetControlledPermanent target = new TargetControlledPermanent(0, 1, filter, true);
+
+        if (!controller.choose(Outcome.Sacrifice, target, source, game)) {
+            return false;
+        }
+
+        Permanent permanent = game.getPermanent(target.getFirstTarget());
+        if (permanent == null) {
+            return false;
+        }
+
+        if (permanent.sacrifice(source, game)) {
+            source.getEffects().setValue("SACRIFICED_PERMANENT", permanent.getId());
```

Comment:
Shouldn't the permanent object itself be stored, so its characteristics can be referenced without checking the game state again?

Reply:
Oh indeed, it is possible to store whole objects.
Was happy enough to store the UUID.